arbor/util/counter.hpp

```diff
@@ -77,17 +78,11 @@ struct counter {
 
     value_type operator[](difference_type n) const { return v_+n; }
 
-    bool operator==(counter x) const { return v_==x.v_; }
-    bool operator!=(counter x) const { return v_!=x.v_; }
-    bool operator<=(counter x) const { return v_<=x.v_; }
-    bool operator>=(counter x) const { return v_>=x.v_; }
-    bool operator<(counter x) const { return v_<x.v_; }
-    bool operator>(counter x) const { return v_>x.v_; }
+    auto operator<=>(const counter& y) const = default;
 
     counter& operator=(const counter&) = default;
     counter& operator=(counter&&) = default;
 
-private:
```

Comment:
why is it needed to make this public?

Reply:
Good question, likely a spurious delete.